app/main.py

```diff
@@ -1,3 +1,54 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if type(other) is Distance:
+            return Distance(
+                self.km + other.km
+            )
+        self.km += other
+        return self
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if type(other) == Distance:
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        self.km = self.km * other
+        return self
+
+    def __truediv__(self, other: int | float | Distance) -> Distance:
+        if type(other) == Distance:
+            Distance(
+                self.km / other.km
+            )
+        self.km = round((self.km / other), 2)
+        return self
```

Comment:
i think here should be new instance too

Reply:
https://stackoverflow.com/questions/1047021/overriding-in-python-iadd-method   We do not return a new one, like in __add__, we overriting using "+=" or similar inside current (self) object.